packages/models/addon/model.js

```diff
@@ -1,3 +1,85 @@
 import DS from 'ember-data';
 import RelationshipTracker from "ember-data-relationship-tracker";
-export default DS.Model.extend(RelationshipTracker);
+import { inject as service } from '@ember/service';
+import { computed, defineProperty, get } from '@ember/object';
+import { readOnly, or } from '@ember/object/computed';
+import { capitalize } from '@ember/string';
+import { run } from '@ember/runloop';
+
+export default DS.Model.extend(RelationshipTracker, {
+  resourceMetadata: service(),
+
+  init() {
+    this._super();
+    let dirtyTrackingProperties = {};
+    this._relationshipsByName().forEach((relation) => {
+      let { kind, meta } = relation;
+      if (meta) {
+        let { owned } = meta.options;
+        if (owned) {
+          let propertyName = createHasDirtyForRelationship(this, meta.name, kind);
+          dirtyTrackingProperties[meta.name] = propertyName;
+        }
+      }
+    });
+    if (Object.keys(dirtyTrackingProperties).length > 0) {
+      createHasDirtyOwned(this, Object.values(dirtyTrackingProperties));
+    }
+    this.set('dirtyTrackingRelationNames', dirtyTrackingProperties);
+  },
+
+  relationshipTrackerVersion: computed(function() {
+    let meta = this.get('resourceMetadata').read(this);
+    return meta && meta.version;
+  }),
+
+  async save() {
+    // this._super is not safe to use asynchronously
+    // see https://github.com/ember-cli/ember-cli/issues/6282
+    let modelSave = this._super.bind(this);
+    run(async () => {
```

Comment:
@ef4 I have no idea why `run` is needed here. The tests that failed because of the lack of it had a `run` wrapped around them but it didn't seem to matter. I think the need for the run loop is going away soon so it didn't seem like a big deal to add this (temporarily?), but let me know you think it's an issue.

Reply:
Yeah, it's because of the async functions. Each time you return from `await` you are starting up with no run loop. It's really only an issue in tests because of the autorun assertion, which is gone in ember 3.4.

You should feel free to update the ember version wherever you see a need for it.